src/read_utils.rs

```diff
@@ -172,7 +174,10 @@ impl ZipHandler {
     }
 }
 
-impl<'a> FileHandler for &'a mut ZipHandler {
+impl<'a, R> FileHandler for &'a mut ZipHandler<R>
+where
+    R: std::io::Seek + std::io::Read,
```

Comment:
You're using multiple times `std::io::Seek` and `std::io::Read` so maybe they can be imported in the prelude of the file with `use` clauses (or maybe just `use std::io;` at least)?

Reply:
I'm not a fan of import I find it important to know where the types are from but since those are quite known Trait I guess we can import them.